carbon-design-system/carbon-addons-iot-react
# feat(datetimepicker): add hasicononly prop

Closes #2492 

**Summary**

- Adds `hasIconOnly` prop to DateTimePicker to optionally only show the icon rather than the selected time range text AND the icon
- Unfortunately this required a fairly large rewrite to use FlyoutMenu as the underlying menu pop out as the current pop out menu was very opinionated and static. It was very difficult to make changes and at the end of the day, FlyoutMenu handles a lot of functionality such as opening the menu in different directions depending on the visibility

**Change List (commits, features, bugs, etc)**

- Added Icon Only story
- Remove div card wrappers for all of the stories as they have no purpose
- Memoize the i18n as we do in other components
- Use FlyoutMenu instead of the custom `expanded` box
- Remove a lot of useless style due to the refactor
- Rewrote most of the tests in react-testing-library and fixed multiple tests that were never working properly (not sure how these ever passed)
- Removed all jest fake timers as they had no effect
- Export `PICKER_KINDS` as we needed them in Monitor and were duplicating the code

**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/1-watson-iot-datetimepicker--icon-only and verify that only the icon appears on first render (check design spec in the linked issue)
- Click the menu and ensure the menu still looks and acts exactly the same as the regular DateTimePicker. This should have NO functional change to the DateTimePicker and should be purely visible
- Use the RTL knob to ensure everything adjusts correctly
- Check that all other stories were unaffected. Also try the `hasIconOnly` knob throughout the stories to ensure everything works as intended

**Regression Test (how to make sure this PR doesn't break old functionality)**

- This should have NO functional change to the DateTimePicker and should be purely visible
